chargeOclock/Source/Models/Client.swift

```diff
@@ -15,10 +15,15 @@ protocol ClientSettable {
 class Client {
 
 	var id: Int = 0
-	var name: String = ""
+	var isDirty: Bool = false
 
-	init(id: Int, name: String) {
-		self.id = id
+	var name: String = "" {
+		didSet {
+			isDirty = true
+		}
+	}
+
+	init(name: String) {
 		self.name = name
```

Comment:
no, the isDirty flag is needed, as it signals whether the data in this object has been changed (a new name was set, for instance)...
the id is a separate issue